### PR TITLE
Detect file changes before run deploy

### PR DESCRIPTION
**Pull request by leorossi**
Mira will craete a `.mira.snapshot` file with an hash of each file of
the project (excluding `node_modules`).
It is based on size and `mtime` rather than content.

If snapshot is the same, mira will not deploy.

Using the flag `--force` mira will ignore snapshot and will run deploy
anyway.

Refs: #21

### CODE REVIEW

File: src/cdk/bootstrap.ts [hunk 1]
```diff
@@ -290,6 +292,8 @@ export class MiraBootstrap {
     }
 
     const cmd = this.args._[0]
+    const cd = new ChangeDetector(process.cwd())
+    const filesChanged = await cd.run()
```
Comment (zimny) [inline src/cdk/bootstrap.ts:296]: there's a small issue that snapshot that is used for the comparison is created in this step, but as CDK runs again later in the code, the modification time is being updated. Effectively it means that deploy will be triggered every time.  

In here https://github.com/nearform/mira/pull/35/files#diff-251cdb280b972bbc33f2c288dc8d52b2R84, the deploy fn was changed to return a promise. When that'll be merged `await cd.takeSnapshot()` should be run after `this.deploy()`. Does it makes sense?

Comment (zimny) [reply src/cdk/bootstrap.ts:299]: well, not sure the exact process, but tested that on the s3 sample app, and deploy was always triggered. Can you check it in the sample app?

Comment (leorossi) [reply src/cdk/bootstrap.ts:299]: I tested on sample app, what I noticed now is that `glob` does not preserve file list, so even with same hashes, generated snapshots are different. I'll fix this ordering the array before writing the file, and I think that will fix the issue.

**Comment by zimny** [comment]
@leorossi there's a conflict, can you solve it?
in general LGTM, but @nigelhanlon could you please test it on s3 sample app if it works as expected now?

**Comment by leorossi** [comment]
Conflict fixed, I tested on S3 during development but a double check is more than helpful!

**Pull request merged**